kong/tools/request_aware_table.lua

```diff
@@ -3,33 +3,27 @@
 
 local table_new = require("table.new")
 local table_clear = require("table.clear")
-local get_request_id = require("kong.tracing.request_id").get
+local request_id = require("kong.tracing.request_id")
+
 
 local is_not_debug_mode = (kong.configuration.log_level ~= "debug")
 
 
 local error        = error
 local rawset       = rawset
 local setmetatable = setmetatable
-local get_phase    = ngx.get_phase
-
-
-local NGX_VAR_PHASES = {
-  set           = true,
-  rewrite       = true,
-  access        = true,
-  content       = true,
-  header_filter = true,
-  body_filter   = true,
-  log           = true,
-  balancer      = true,
-}
+
+
+local get_request_id = request_id.get
+local var_available = request_id.var_available
+
+
 local ALLOWED_REQUEST_ID_K = "__allowed_request_id"
 
 
 -- Check if access is allowed for table, based on the request ID
 local function enforce_sequential_access(table)
-  if not NGX_VAR_PHASES[get_phase()] then
+  if not var_available() then
```

Comment:
I am not sure about this line, could we use:
```
local curr_request_id = get_request_id()
if not curr_request_id then
...
```

Reply:
yes, I think we can do as you say and avoid the check here. Request-aware-table was implemented before the `request_id` feature existed, so as you noticed there are a couple of things that can be improved.

Reply:
Thank you @samugi , I have change the code, please review it again.